build-scripts/xc-universal-binary.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+set -euvx
+
+# This should be invoked from inside xcode, not manually
+if [ "$#" -ne 4 ]
+then
+    echo "Usage (note: only call inside xcode!):"
```

Comment:
Additionally, we could test for the `ACTION` env variable to detect if we are running in a XCode context and print a big warning message/abort otherwise.

Reply:
Eh, if someone wants to run this manually they can, I just wanted to make it clear that they were no longer following the intended use case.